modules/vulnix.nix

```diff
@@ -0,0 +1,183 @@
+          })} \
+            --cache-dir $CACHE_DIRECTORY \
+            ${lib.concatStringsSep " " cfg.extraOpts} "$@" \
+          || case $? in
+            0 ) ;; # no vulnerabilities found
+            1 ) ;; # only whitelisted vulnerabilities found
+            2 ) ;; # vulnerabilities found
+            * ) exit $? ;; # unexpected
+          esac
+        }
+
+        scan ${lib.cli.toGNUCommandLineShell {} (with cfg; {
+          system = scanSystem;
+          gc-roots = scanGcRoots;
+        })} \
+          -- ${lib.escapeShellArgs cfg.paths} \
+        | ${cfg.sink}
+      '' + lib.optionalString cfg.scanNomadJobs ''
+        export VAULT_TOKEN=$(< $CREDENTIALS_DIRECTORY/vault-token)
+        NOMAD_TOKEN=$(vault read -format json -field secret_id nomad/creds/admin | jq -rj)
+        sleep 5s # let nomad token be propagated to come into effect
+
+        [[ -f $STATE_DIRECTORY/index ]] || {
+          printf '%d' 0 > $STATE_DIRECTORY/index
+        }
+
+        # TODO If the NOMAD_TOKEN expires the service would probably exit uncleanly and restart. Make it a clean restart.
+
+        # TODO get rid of function and use only one curl invocation
+        # with `namespace=*` once we get https://github.com/hashicorp/nomad/pull/10935
+        function stream {
+          <<< X-Nomad-Token:"$NOMAD_TOKEN" \
+          curl -H @- \
+            --no-progress-meter \
+            --cacert /etc/ssl/certs/ca.pem \
+            -NG "$NOMAD_ADDR"/v1/event/stream \
+            --data-urlencode namespace="$1" \
+            --data-urlencode topic=Job \
+            --data-urlencode index=$(< $STATE_DIRECTORY/index) \
+          | jq --unbuffered -rc 'select(length > 0) | {"index": .Index} as $out | .Events[] | select(.Type == "EvaluationUpdated").Payload.Job | $out * {"namespace": .Namespace, "job": .ID} as $out | .TaskGroups[] | $out * {"taskgroup": .Name} as $out | .Tasks[] | $out * {"task": .Name, "flake": .Config.flake}' \
+          | while read -r job; do
+            <<< "$job" jq -rc .flake \
+            | XDG_CACHE_HOME=$CACHE_DIRECTORY \
+              GIT_SSH_COMMAND="${pkgs.openssh}/bin/ssh -i $CREDENTIALS_DIRECTORY/ssh" \
+              xargs -L 1 \
+              nix --netrc-file $CREDENTIALS_DIRECTORY/netrc show-derivation \
+            | jq --unbuffered -r keys[] \
+            | while read -r drv; do
+              scan -- "$drv" \
+              | NOMAD_JOB_NAMESPACE=$(<<< "$job" jq -rj .namespace) \
+                NOMAD_JOB_ID=$(<<< "$job" jq -rj .job) \
+                NOMAD_JOB_TASKGROUP_NAME=$(<<< "$job" jq -rj .taskgroup) \
+                NOMAD_JOB_TASK_NAME=$(<<< "$job" jq -rj .task) \
+                ${cfg.sink}
+            done
+            <<< "$job" jq -r .index > $STATE_DIRECTORY/index
+          done
+        }
+        stream default &
+        stream midnight-testnet &
+        stream midnight-unstable &
+        stream midnight-benchmarking &
+        stream midnight-qa &
+        stream midnight-urs &
```

Comment:
Right, confused repos... I only hard coded these [temporarily](https://github.com/input-output-hk/bitte/pull/63/commits/67588ef47709094980ea38b0d95fa7037457153b#diff-db98b3d946665227fb7380fc013babe05aab0053c30d9b97ece1c2f96e045edfR130) because of https://github.com/hashicorp/nomad/pull/10935. A config option would be nicer than just listening to all namespaces though